Sources/Classes/core/Storage/MiniAppSecureStorage.swift

```diff
@@ -190,30 +190,73 @@ public class MiniAppSecureStorage: MiniAppSecureStorageDelegate {
         MiniAppLogger.d("🔑 Secure Storage: write store to disk completed")
     }
 
-    // MARK: - Clear
-    public static func wipeSecureStorages() throws {
+    // MARK: - Clear All Secure Storage
+    internal static func wipeSecureStorages() {
         MiniAppLogger.d("🔑 Secure Storage: destroy")
-        let cachePath = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)[0]
-        let miniAppPath = cachePath.appendingPathComponent("/MiniApp/")
-        guard let contentNames = try? FileManager.default.contentsOfDirectory(atPath: miniAppPath.path) else { return }
+        guard let contentNames = try? FileManager.default.contentsOfDirectory(atPath: FileManager.getMiniAppFolderPath().path) else { return }
         for name in contentNames {
-            let url = miniAppPath.appendingPathComponent("/" + name)
-            if let isDirectory = (try url.resourceValues(forKeys: [.isDirectoryKey])).isDirectory, isDirectory {
+            let url = FileManager.getMiniAppFolderPath().appendingPathComponent("/" + name)
+            do {
+                if let isDirectory = (try url.resourceValues(forKeys: [.isDirectoryKey])).isDirectory, isDirectory {
+                    do {
+                        try FileManager.default.removeItem(at: url.appendingPathComponent("/" + MiniAppSecureStorage.storageFullName))
+                        MiniAppLogger.d("🔑 Secure Storage: destroyed storaged for \(name)")
+                    } catch {
+                        MiniAppLogger.d("🔑 Secure Storage: could not destroy storaged for \(name)")
+                    }
+                } else {
+                    MiniAppLogger.d("🔑 Secure Storage: ignored \(name)")
+                }
+            } catch let error {
+                MiniAppLogger.d("🔑 Secure Storage Wipe Failed: \(name)", error.localizedDescription)
+            }
+        }
+    }
+
+    // MARK: - Clear storage for MiniApp ID
+    internal static func wipeSecureStorage(for miniAppId: String) {
+        if !miniAppId.isEmpty {
```

Comment:
just a question regarding this, the secure storage file will always be located in the miniapp directory so all versions can access it + the filename doesn't change so I don't think it's necessary to traverse or am I missing sth?

Reply:
You are right man. Thanks! Updated them now